zookeeper-server/src/main/java/org/apache/zookeeper/server/util/DigestCalculator.java

```diff
@@ -35,18 +35,21 @@
 
     // The hardcoded digest version, should bump up this version whenever
     // we changed the digest method or fields.
-    //
-    // Defined it as Integer to make it able to be changed in test via reflection
-    public static final Integer DIGEST_VERSION = 2;
+    private static final int DIGEST_VERSION = 2;
 
     public static final String ZOOKEEPER_DIGEST_ENABLED = "zookeeper.digest.enabled";
-    private static boolean digestEnabled;
 
     static {
-        digestEnabled = Boolean.parseBoolean(System.getProperty(ZOOKEEPER_DIGEST_ENABLED, "true"));
-        LOG.info("{} = {}", ZOOKEEPER_DIGEST_ENABLED, digestEnabled);
+        LOG.info("{} = {}", ZOOKEEPER_DIGEST_ENABLED, System.getProperty(ZOOKEEPER_DIGEST_ENABLED, "true"));
     }
 
+    private boolean digestEnabled;
+
+    public DigestCalculator() {
```

Comment:
Move it to the same package as `DataTree` lives and make package-private everything.

Reply:
@lvfangmin mentioned that he will use the DigestCalculator from other places, so I would wait for his PR. If it turns out still to be used only from this package, then I will move it.

Reply:
He mentioned `PrepRequestProcessor` which is also in `server` package, so it should be even better there.

Reply:
Let's make the class 'final' it will slightly help the JVM for optimizations.
We had 'static' methods that were not overridable.

Reply:
This is a good idea. I tried it, but in `SnapshotDigestTest.testDifferentDigestVersion` we create a Mockito Spy on the class and apparently you can not create Mockito Spy on a final class.

So instead of this I take @anmolnar 's original idea and moved this class to the `server` package and made it packet private, so at least it can not be overwritten from non-ZK code. Let's hope it will work also for @lvfangmin. If not, then we can move back...

Reply:
@lvfangmin I think that `DigestCalculator` is an internal utility of `DataTree` and should not be exposed to anywhere else. `PrepRequestProcessor` could do better to access properties like `digestEnabled` and `DIGEST_VERSION` through `DataTree` getters. Like:
```java
zks.getZKDatabase().getDataTree().getDigestEnabled();
```
It can cache the 2 values in the constructor like it's doing currently with `digestEnabled`.

Unfortunately `SerializeUtils` is a different story. It's another piece of logic which I think should be organised differently. It breaks both encapsulation and responsibility principles. Let me think about if there's a way to get round that.